p2p/listener.go

```diff
@@ -52,8 +52,8 @@ func newListenersP2P(host p2phost.Host) *Listeners {
 		defer reg.RUnlock()
 
 		l := reg.Listeners[stream.Protocol()]
-		if l != nil {
-			go l.(*remoteListener).handleStream(stream)
```

Comment:
triage notes: @gitsrc  when was this causing panic? fixing the underlying issue sounds like the right way, we dont want to silence the error

Reply:
Thanks for the note! To clarify, no panic has actually occurred. While reading and learning this part of the code, I felt it could use some additional robustness to prevent potential issues.

Reply:
Hm.. for sure we don't want to hide the problem if it ever occurs, as that will be painful to debug.

I'm leaning towards closing this PR unless convinced there is net value add in doing this (panic repro or description of why its ok to ignore panic).

Reply:
OK, got it.